src/sst/core/config.cc

```diff
@@ -41,163 +43,826 @@ using namespace std;
 #ifdef __SST_DEBUG_EVENT_TRACKING__
-    event_dump_file = "";
+    event_dump_file_ = "";
 #endif
 
-    // Some config items can be initialized from either the command line or
-    // the config file. The command line has precedence. We need to initialize
-    // the items found in the sdl file first. They will be overridden later.
-    // We need to find the configFile amongst the command line arguments
+    // Advanced Options- environment
+    print_env_           = false;
+    enable_sig_handling_ = true;
+    no_env_config_       = false;
+}
+
+
+// Check for existence of config file
+bool
+Config::checkConfigFile()
+{
+    struct stat sb;
+    char*       fqpath = realpath(configFile_.c_str(), nullptr);
+    if ( nullptr == fqpath ) {
+        fprintf(stderr, "Failed to canonicalize path [%s]:  %s\n", configFile_.c_str(), strerror(errno));
+        return false;
+    }
+    configFile_ = fqpath;
+    free(fqpath);
+    if ( 0 != stat(configFile_.c_str(), &sb) ) {
+        fprintf(stderr, "File [%s] cannot be found: %s\n", configFile_.c_str(), strerror(errno));
+        return false;
+    }
+    if ( !S_ISREG(sb.st_mode) ) {
+        fprintf(stderr, "File [%s] is not a regular file.\n", configFile_.c_str());
+        return false;
+    }
+
+    if ( 0 != access(configFile_.c_str(), R_OK) ) {
+        fprintf(stderr, "File [%s] is not readable.\n", configFile_.c_str());
+        return false;
+    }
+
+    return true;
+}
+
+std::string
+Config::getLibPath(void) const
+{
+    char* envpath = getenv("SST_LIB_PATH");
+
+    // Get configuration options from the user config
+    std::vector<std::string>                          overrideConfigPaths;
+    SST::Core::Environment::EnvironmentConfiguration* envConfig =
+        SST::Core::Environment::getSSTEnvironmentConfiguration(overrideConfigPaths);
+
+    std::string           fullLibPath  = libpath_;
+    std::set<std::string> configGroups = envConfig->getGroupNames();
+
+    // iterate over groups of settings
+    for ( auto groupItr = configGroups.begin(); groupItr != configGroups.end(); groupItr++ ) {
+        SST::Core::Environment::EnvironmentConfigGroup* currentGroup = envConfig->getGroupByName(*groupItr);
+        std::set<std::string>                           groupKeys    = currentGroup->getKeys();
+
+        // find which keys have a LIBDIR at the END of the key
+        // we recognize these may house elements
+        for ( auto keyItr = groupKeys.begin(); keyItr != groupKeys.end(); keyItr++ ) {
+            const std::string key   = *keyItr;
+            const std::string value = currentGroup->getValue(key);
+
+            if ( key.size() > 6 ) {
+                if ( "LIBDIR" == key.substr(key.size() - 6) ) {
+                    fullLibPath.append(":");
+                    fullLibPath.append(value);
+                }
+            }
+        }
+    }
+
+    // Clean up and delete the configuration we just loaded up
+    delete envConfig;
+
+    if ( nullptr != envpath ) {
```

Comment:
We'll need to as @nmhamster about the code in this function.  I'm not entirely sure what it's doing and since there are not obvious bugs, I'm not going to take the time to figure it out.

Reply:
@calewis - yes to checking first, this is all in initialization and very cheap. I think way back we had some other code in here. But yes, you can check early and not build from configuration if we are supplied in the environment or command line (these are the two `if` checks). This allows environment variables to override configuration and then command line to override environment and configuration.